app-backend/tile/src/main/scala/LayerCache.scala

```diff
@@ -175,40 +175,44 @@ object LayerCache extends Config with LazyLogging with KamonTrace {
-        oldAst   <- OptionT.fromOption[Future]({
-          logger.debug(s"Parsing Tool AST with ${tool.definition}")
-          val entireAST = tool.definition.as[MapAlgebraAST].valueOr(throw _)
-          subNode.flatMap(id => entireAST.find(id)).orElse(Some(entireAST))
-        })
-        subs     <- assembleSubstitutions(oldAst, { id: UUID =>
-          OptionT(Tools.getTool(id, user))
-            .map({ referrent => referrent.definition.as[MapAlgebraAST].valueOr(throw _) })
-            .value
-        })
-        ast      <- OptionT.fromOption[Future](oldAst.substitute(subs))
-        nodeId   <- OptionT.pure[Future, UUID](subNode.getOrElse(ast.id))
-        params   <- OptionT.pure[Future, EvalParams]({
-          logger.debug(s"Parsing ToolRun parameters with ${toolRun.executionParameters}")
-          val parsedParams = toolRun.executionParameters.as[EvalParams].valueOr(throw _)
-          val defaults = ast.metadata
-          val overrides = parsedParams.metadata.get(ast.id)
-          val md = (overrides |@| defaults).map(_.fallbackTo(_))
-            .orElse(overrides)
-            .orElse(defaults)
-            .getOrElse(NodeMetadata())
-
-          EvalParams(
-            parsedParams.sources,
-            parsedParams.metadata + (ast.id -> md)
-          )
-        })
-      } yield (ast, params)
+  ): OptionT[Future, (MapAlgebraAST, EvalParams)] =
+    traceName("LayerCache.toolEvalRequirements") {
+      val cacheKey = s"ast+params-$toolRunId-${subNode}-${user.id}"
+      if (voidCache) histogramCache.delete(cacheKey)
+      astCache.cachingOptionT(cacheKey) { implicit ec =>
+        traceName("LayerCache.toolEvalRequirements (no cache)") {
+          for {
+            (tool, toolRun) <- LayerCache.toolAndToolRun(toolRunId, user)
+            oldAst   <- OptionT.fromOption[Future]({
+              logger.debug(s"Parsing Tool AST with ${tool.definition}")
+              val entireAST = tool.definition.as[MapAlgebraAST].valueOr(throw _)
+              subNode.flatMap(id => entireAST.find(id)).orElse(Some(entireAST))
+            })
+            subs     <- assembleSubstitutions(oldAst, { id: UUID =>
+              OptionT(Tools.getTool(id, user))
+                .map({ referrent => referrent.definition.as[MapAlgebraAST].valueOr(throw _) })
+                .value
+            })
+            ast      <- OptionT.fromOption[Future](oldAst.substitute(subs))
+            nodeId   <- OptionT.pure[Future, UUID](subNode.getOrElse(ast.id))
+            params   <- OptionT.pure[Future, EvalParams]({
+              logger.debug(s"Parsing ToolRun parameters with ${toolRun.executionParameters}")
+              val parsedParams = toolRun.executionParameters.as[EvalParams].valueOr(throw _)
+              val mdOverride = parsedParams.metadata.get(ast.id)
+              val md = (mdOverride |@| ast.metadata).map(_.fallbackTo(_))
+                .orElse(mdOverride)
+                .orElse(ast.metadata)
```

Comment:
Nothing too edifying to say here - mainly some indentation cleanup (which is why the diff is nuts) and relabelling of `overrides` to `mdOverride` due to override being ambiguous between overriding a portion of the tree and overriding a node's metadata